main.js

```diff
@@ -200,24 +206,24 @@ function logRedirect(isActive, id, reason) {
 
 function createStates(onConnect) {
     const _inst = new States({
-        namespace: 'host.' + hostname,
+        namespace: hostLogPrefix,
         connection: config.states,
         logger: logger,
         hostname: hostname,
         change: (id, state) => {
             inputCount++;
             if (!id) {
-                logger.error('host.' + hostname + ' change event with no ID: ' + JSON.stringify(state));
+                logger.error(hostLogPrefix + ' change event with no ID: ' + JSON.stringify(state));
                 return;
             }
             // If some log transporter activated or deactivated
             if (id.match(/.logging$/)) {
                 logRedirect(state ? state.val : false, id.substring(0, id.length - '.logging'.length), id);
             } else
             // If this is messagebox
-            if (id === 'messagebox.system.host.' + hostname) {
+            if (!compactGroupController && id === 'messagebox.' + hostObjectPrefix) {
```

Comment:
I'm not very deep into the inner workings of this. Why is this not allowed for compactGroupController ?

Reply:
because only the "real" host controller is handling messages (execure commands for this host and such) and so only relevant if "not" a compact group controller